### PR TITLE
Make school zone beacons from open dataset available to choose in components form 

### PR DESCRIPTION
**Pull request by chiaberry**
## Associated issues

https://github.com/cityofaustin/atd-data-tech/issues/18533

## Testing
**URL to test:** 

Locally

**Steps to test:**

Add a new component to the map,  component type Signal - School Zone Beacon. There will be an autocomplete that appears once youve chosen that component type. 

Screenshot 2024-09-17 at 10 47 37 AM

you can search for a specific beacon, or scroll to find one. Note that the knack/socrata dataset inserts extra spaces between the words, so if you are looking for `1406 Dale` you will need to search `1406  Dale`

Save the component. Much like when adding a signal, the create component flow bypasses the manually mapping step and zooms the map to what has been added. 

Test editing an existing Beacon, confirm the correct beacon appears in the autocomplete. Only change the attributes. Save. 
Test editing an existing Beacon by switching it to a Signal. Confirm the old beacon does not appear on the map and is deleted when checking the table
```
SELECT * FROM feature_school_beacons fsb 
```
Test switching a signal to a beacon.

Test deleting the component. The beacon is no longer associated with the project. 

---
#### Ship list
- [ ] Code reviewed 
- [ ] Product manager approved
- [ ] Product manager checked [DB view dependencies](https://atd-dts.gitbook.io/moped-documentation/product-management/updating-the-arcgis-online-components-database-view)
- [ ] Product manager added to [QA test script](https://docs.google.com/spreadsheets/d/1n_O6MLh9cwwPf57HUM394Ea-z9uuoEV1-QW4axNZXLE/edit#) if applicable

### CODE REVIEW

File: moped-editor/src/views/projects/projectView/ProjectComponents/SchoolBeaconComponentAutocomplete.js
Comment (johnclary) [inline moped-editor/src/views/projects/projectView/ProjectComponents/SchoolBeaconComponentAutocomplete.js:1]: not 100% sure on this or how much we care, but it seams like the signal and school beacon autocompletes could be one component with diff props 👀

**Review by johnclary — CHANGES_REQUESTED** [review]
Wow—this looks great 🤩 

~I am running into one issue which occurs in a few different edit scenarios~

Update: i am no longer able to reproduce this bug 🙃 

This feature is looking awesome—you've done a great job working this into the existing code 🔥

File: moped-database/migrations/1725983667950_add_beacons_asset_table/up.sql [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    beacon_id text,
+    knack_id text NOT NULL,
+    location_name text,
+    zone_name text,
+    beacon_name text,
+    geography geography('MULTIPOINT') NOT NULL,
+    created_at timestamptz NOT NULL DEFAULT now(),
+    created_by_user_id int4 NULL,
+    updated_by_user_id int4 NULL,
+    updated_at timestamptz NULL
```
[the remark this answers is not in the document]
Comment (mddilley) [reply moped-database/migrations/1725983667950_add_beacons_asset_table/up.sql:11]: good catch! yes, could you make that update @chiaberry? And, while you are at it, could you copypasta the comments that we have on these four audit fields in the other feature tables? thank you! 🙏 🙏

File: moped-editor/src/views/projects/projectView/ProjectComponents/SchoolBeaconComponentAutocomplete.js
Comment (chiaberry) [reply moped-editor/src/views/projects/projectView/ProjectComponents/SchoolBeaconComponentAutocomplete.js:1]: ill try it!!

File: moped-editor/src/views/projects/projectView/ProjectComponents/ProjectComponentsListItem.js [hunk 2]
```diff
@@ -116,7 +117,7 @@ const ProjectComponentsListItem = ({
 
           <Tooltip
             title={
-              isSignal
+              isSignal || isSchoolBeacon
                 ? "Signal locations can only be changed by editing the component details"
                 : "Map"
```
[the remark this answers is not in the document]
Comment (chiaberry) [reply moped-editor/src/views/projects/projectView/ProjectComponents/ProjectComponentsListItem.js:122]: updated! thank you for the rewording

**Review by mddilley — COMMENTED** [review]
Unleashing some comments that I've been holding on to this afternoon. I'll keep reviewing! It is looking ✨!

File: moped-database/migrations/1725983667950_add_beacons_asset_table/up.sql [hunk 3]
```diff
@@ -0,0 +1,23 @@
+CREATE TABLE public.feature_school_beacons (
+    beacon_id text,
```
[the remark this answers is not in the document]
Comment (chiaberry) [reply moped-database/migrations/1725983667950_add_beacons_asset_table/up.sql:2]: I cant believe I missed this, I am requesting the field in the socrata endpoint, but not saving it in the database. 

I am using the beacon_id in order to create the unique option label, but the school_zone_beacon_id is what I should use as the layers primary key column

File: moped-editor/src/utils/signalComponentHelpers.js [hunk 4]
```diff
@@ -189,3 +195,68 @@ export const generateProjectComponent = (
...
+  if (schoolBeacon && schoolBeacon?.properties && schoolBeacon?.geometry) {
+    const featureSchoolBeaconsRecord = {
+      // MultiPoint coordinates are an array of arrays, so we wrap the coordinates
+      geography: {
+        ...schoolBeacon.geometry,
+        type: "MultiPoint",
+        coordinates: [schoolBeacon.geometry.coordinates],
+      },
+      knack_id: schoolBeacon.properties.id,
+      location_name: `${schoolBeacon.properties.zone_name} ${schoolBeacon.properties.beacon_name}`,
```
[the remark this answers is not in the document]
Comment (chiaberry) [reply moped-editor/src/utils/signalComponentHelpers.js:233]: I am going to make this change, but now I am wondering if we need the zone name or beacon name from knack at all....

[the remark this answers is not in the document]
Comment (chiaberry) [reply moped-editor/src/utils/signalComponentHelpers.js:233]: never mind! we still need them in the options formatter for the auto complete

File: moped-editor/src/views/projects/projectView/ProjectComponents/KnackComponentAutocomplete.js [hunk 5]
```diff
@@ -45,18 +60,18 @@ const SignalComponentAutocomplete = React.forwardRef(
...
-        <Alert severity="error">{`Unable to load signal list: ${error}`}</Alert>
+        <Alert severity="error">{`Unable to load ${componentLabel} list: ${error}`}</Alert>
       );
     }
 
     return (
       <Autocomplete
         className={classes}
-        id="signal-id"
+        id="knack-component-autocomplete-id"
```
Comment (mddilley) [inline moped-editor/src/views/projects/projectView/ProjectComponents/KnackComponentAutocomplete.js:70]: @chiaberry Could you remove the `-id` part of this id string? This is causing the 1PW extension to want to autofill on it for some reason. 🤷 This is an existing issue in prod for this autocomplete signals so not something you introduced.

Screenshot 2024-09-23 at 6 08 01 PM

File: moped-editor/src/views/projects/projectView/ProjectComponents/utils/makeComponentData.js [hunk 6]
```diff
@@ -147,6 +162,41 @@ export const getFeatureChangesFromComponentForm = (
       // signal was previously blank
       signalToCreate = knackSignalRecordToFeatureSignalsRecord(signalFromForm);
       signalToCreate.component_id = clickedComponent.project_component_id;
+      // if there was a beacon that was switched to signal, we need to clear that beacon
```
Comment (mddilley) [inline moped-editor/src/views/projects/projectView/ProjectComponents/utils/makeComponentData.js:165]: This is also a good reminder of how we handle all of the update scenarios that can happen through the form!

File: moped-database/migrations/1725983667950_add_beacons_asset_table/up.sql (hunk 1 above)
[the remark this answers is not in the document]
Comment (mddilley) [reply moped-database/migrations/1725983667950_add_beacons_asset_table/up.sql]: @chiaberry could you add the comments on the audit fields `created_at`, `created_by_user_id`, `updated_by_user_id`,`updated_at`? The same ones that we have on the other feature tables. Thanks for updating the update_at default! 🙏

File: moped-database/migrations/1725983667950_add_beacons_asset_table/up.sql [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+    knack_id text NOT NULL,
+    location_name text,
+    zone_name text,
+    beacon_name text,
```
[the remark this answers is not in the document]
Comment (mddilley) [reply moped-database/migrations/1725983667950_add_beacons_asset_table/up.sql]: thanks for updating, Chia! 🙏

**Review by mddilley — CHANGES_REQUESTED** [review]
This looks ready to go to me! I mentioned you in a couple comments for some last small changes. One was from John's review. I think this is ready to go after that and when the next two issues are complete. 🙌

File: moped-editor/src/views/projects/projectView/ProjectComponents/CreateComponentModal.js [hunk 8]
```diff
@@ -10,7 +10,7 @@ const CreateComponentModal = ({
   onSaveDraftSignalComponent,
 }) => {
   const onSave = (formData) => {
-    const isSavingSignalFeature = Boolean(formData.signal);
+    const isSavingSignalFeature = Boolean(formData.signal || formData.schoolBeacon);
```
[the remark this answers is not in the document]
Comment (chiaberry) [reply moped-editor/src/views/projects/projectView/ProjectComponents/CreateComponentModal.js]: its a good detail.

**Comment by roseeichelmann** [comment]
I'm getting a bit confused by `Signal - School Zone Beacon`, `Flashing Zone Beacon -` , and `Signal - PHBs`. In your instructions when you say test editing a Beacon, do you mean a `Signal - School Zone Beacon`? And what do you mean by test changing it to a signal? Do you want us to update it to like `Signal - PHB`? And vice versa? Or do i have that totally wrong. Im not super familiar with the different component terminology so im getting so confused by the overlapping names

**Comment by chiaberry** [comment]
@roseeichelmann yeah sorry, the nomenclature is pretty confusing and I totally forgot about the Flashing Zone Beacon. 

School Zone Beacon is what I mean when I say "Test the beacons" (in the future its going to be renamed to Beacon -School Zone so hopefully that helps https://github.com/cityofaustin/atd-data-tech/issues/19040)

and the change to a signal, change it to either a Signal PHB or Signal Traffic

File: moped-database/migrations/1725983667950_add_beacons_asset_table/up.sql (hunk 1 above)
[the remark this answers is not in the document]
Comment (mddilley) [reply moped-database/migrations/1725983667950_add_beacons_asset_table/up.sql]: looks great! thank you Chia!

File: moped-editor/src/views/projects/projectView/ProjectComponents/KnackComponentAutocomplete.js (hunk 5 above)
Comment (mddilley) [reply moped-editor/src/views/projects/projectView/ProjectComponents/KnackComponentAutocomplete.js]: yay! thank you for squeezing this in. Such a nice detail to tie up 🙌

**Review by mddilley — APPROVED** [review]
I tested this again through all the ways of creating a school zone beacon, and it all looks great to me! Putting this approval on here so it will be ready for when the time comes to merge with the other branches. 🙌

**Review by johnclary — APPROVED** [review]
i tested this a bunch more—it works great. thanks for making those changes. this is exciting! 🚢 🚢 🚢

**Pull request merged**